### PR TITLE
benchmarks for afsa

### PR DESCRIPTION
**Pull request by Fruktus**
- added benchmarks for all implemented test functions
- added custom comparator to afsa
- added verbose option

~~check if multiprocessing needs to be added to requirements~~ mp is part of standard lib

### CODE REVIEW

File: artifishswarm/afsa.py [hunk 1]
```diff
@@ -84,12 +90,13 @@ def prey(self, fish_idx: int):
...
         for _ in range(self.search_retries):
-            target_x = self.fish[fish_idx] + self.vision * self.rng.uniform(-1, 1)
+            target_x = self.fish[fish_idx] + \
+                self.vision * self.rng.uniform(-1, 1)
             # originally the random is between [0,1) but it may be worth exploring
             # how it behaves when allowed the range [-1,1]
 
             target_y = self.func(target_x)
-            if target_y > self.func(self.fish[fish_idx]):
+            if self.comparator(target_y, self.func(self.fish[fish_idx])):
```
Comment (kjarosh) [inline artifishswarm/afsa.py:99]: Normally, comparator returns a value > 0 if the first argument is greater than the second, < 0 if the first is less than the second, and 0 if they are equal. I would suggest applying these semantics here.

File: artifishswarm/afsa.py [hunk 2]
```diff
@@ -249,10 +264,10 @@ def run(self) -> None:
 
         self.history = pd.DataFrame()
         self.best_x = None
-        self.best_y = float('-inf')
+        self.best_y = float('-inf') if self.comparator(float('inf'), float('-inf')) else float('inf')
```
Comment (kjarosh) [inline artifishswarm/afsa.py:267]: Why not define the worst value in the constructor once and reassign it here?

File: examples/benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        optimum_y.append(afsa.result[1])
+
+    return optimum_x, optimum_y
+
+
+def worker(job_params):
+    func_name = job_params.pop('func_name')
+
+    res_x, res_y = run_experiment(job_params)
+    print(f'{func_name} best_x: {np.average(res_x, axis=0)} std: {np.std(res_x, axis=0)}, best_y: {np.average(res_y)} std: {np.std(res_y)} actual_y: 0.0')
```
Comment (kjarosh) [inline examples/benchmarks.py:51]: Actual best Y is always 0, but it's not the case for e.g. eggholder

Comment (Fruktus) [reply examples/benchmarks.py]: Added default values

**Pull request merged**